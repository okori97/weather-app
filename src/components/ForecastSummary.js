import React from "react";
import iconData from "../data/iconData.json";
import "../styles/ForecastSummary.css";
import { GetDate } from "../helpers/dateUtils";
import addSuffix from "../helpers/addSuffix";

export default function ForecastSummary(props) {
	const { date, description, temperature, icon, onSelect } = props;
	const iconCode = icon.toString().slice(0, 1) + "00";

	// Converts unix into date and adds suffix to month
	const { day, monthNum, monthString } = GetDate(date);
	let convertedDate = `${day} ${addSuffix(monthNum)} ${monthString}`;

	return (
		<div className="forecast-summary" data-testid="forecast-summary">
			<div className="forecast-summary__date">{convertedDate}</div>
			<img
				className="forecast-summary__icon"
				data-testid="forecast-icon"
				src={iconData[iconCode]}
				alt="weather icon"
			></img>
			<div className="forecast-summary__temperature">{`${temperature.max}Â°c`}</div>
			<div className="forecast-summary__description">{description}</div>
			<button
				type="button"
				className="forecast-summary__button"
				onClick={() => onSelect(date)}
			>
				More details
			</button>
		</div>
	);
}
