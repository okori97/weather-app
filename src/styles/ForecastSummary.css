.forecast-summary {
  padding: 24px 16px 16px 0;
  font-size: 15px;
}

.forecast-summary__date {
  padding: auto;
}
.forecast-summary__description {
  padding: auto;
}
.forecast-summary__icon {
  padding: 8px 0px 8px 0px;
}
.forecast-summary__temperature {
  padding: auto;
}

.forecast-summary__link-wrap {
  margin-top: 4px;
}
