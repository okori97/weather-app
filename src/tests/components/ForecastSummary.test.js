/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1111111,
    description: "stubDescription",
    icon: 800,
    temperature: {
      max: 11,
      min: 4,
    },
    onSelect: () => {},
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
        onSelect={validProps.onSelect}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct value for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon.toString()}
        temperature={validProps.temperature}
      />
    );

    expect(getByText("Thu 1st Jan")).toHaveAttribute(
      "class",
      "forecast-summary__date"
    );
    expect(getByText("stubDescription")).toHaveClass(
      "forecast-summary__description"
    );
    expect(getByText("11Â°c")).toHaveClass("forecast-summary__temperature");
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary__icon");
  });
});
