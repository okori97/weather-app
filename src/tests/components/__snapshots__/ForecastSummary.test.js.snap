// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
      Thu 1st Jan
    </div>
    <img
      alt="weather icon"
      class="forecast-summary__icon"
      data-testid="forecast-icon"
      src="https://openweathermap.org/img/wn/01d@2x.png"
    />
    <div
      class="forecast-summary__temperature"
    >
      11Â°c
    </div>
    <div
      class="forecast-summary__description"
    >
      stubDescription
    </div>
    <div
      class="forecast-summary__link-wrap"
    >
      <button
        class="forecast-summary__link"
        href="#"
        target="blank"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
